### PR TITLE
Node type references

### PR DESCRIPTION
**Pull request by skurfuerst**

### CODE REVIEW

**Comment by skurfuerst** [comment]
marz-05-2017 11-28-10

File: src/main/java/de/vette/idea/neos/lang/yaml/references/nodeType/NodeTypeReference.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+
+    NodeTypeReference(YAMLKeyValue yamlElement) {
+        // the "textRange" is used for highlighting the source element
+        super(yamlElement, new TextRange(0, yamlElement.getKey().getTextLength() - 1));
+        this.yamlElement = yamlElement;
+    }
+
+    @NotNull
+    @Override
+    public Object[] getVariants() {
```
Comment (Haehnchen) [inline src/main/java/de/vette/idea/neos/lang/yaml/references/nodeType/NodeTypeReference.java]: no todo :) deprecated use GotoCompletionRegistrar

File: src/main/java/de/vette/idea/neos/indexes/NodeTypesYamlFileIndex.java [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+                    .forEach(yamlKeyValue -> result.put(yamlKeyValue.getKeyText(), null));
+            return result;
+        }
+
+        private static boolean isValidForIndex(FileContent inputData, PsiFile psiFile) {
+            if (inputData.getFile().getLength() > MAX_FILE_BYTE_SIZE) {
+                return false;
+            }
+
+            return psiFile.getName().startsWith("NodeTypes.") && psiFile.getName().endsWith(".yaml");
```
Comment (Haehnchen) [inline src/main/java/de/vette/idea/neos/indexes/NodeTypesYamlFileIndex.java]: `yaml` and `yml` are registered as YamlFileTypes not need for extension check

File: src/main/java/de/vette/idea/neos/lang/yaml/references/nodeType/NodeTypeReference.java (hunk 1 above)
Comment (skurfuerst) [inline src/main/java/de/vette/idea/neos/lang/yaml/references/nodeType/NodeTypeReference.java]: What do you mean? :) Could you explain some more please? Thanks <3

File: README.md [hunk 3]
```diff
@@ -28,15 +28,26 @@ You can install the plugin by going to `Settings > Plugins > Browse repositories
...
+  - right-click it and select "Run"
+  - (you can also select "debug" to run it in debug mode!)
+- in debug mode, you can hot-reload classes by choosing "Run -> Reload changed classes" as explained on https://www.jetbrains.com/help/idea/2016.3/reloading-classes.html
 - have fun developing :)
 
 Development Tips and Tricks
 ---------------------------
 
 - install GrammarKit if you want to work on the parser
+- to log, use `com.intellij.openapi.diagnostic.Logger.getInstance("!!! CUSTOM").warn("FOO");`
```
Comment (skurfuerst) [inline README.md]: Thanks <3 will cahnge

File: src/main/java/de/vette/idea/neos/indexes/NodeTypesYamlFileIndex.java (hunk 2 above)
Comment (skurfuerst) [inline src/main/java/de/vette/idea/neos/indexes/NodeTypesYamlFileIndex.java]: good point; will remove!

File: src/main/java/de/vette/idea/neos/lang/yaml/references/nodeType/NodeTypeReference.java (hunk 1 above)
Comment (Haehnchen) [inline src/main/java/de/vette/idea/neos/lang/yaml/references/nodeType/NodeTypeReference.java]: `getVariants()` was not designed for a lookup elements implementation. i was told that this feature will be dropped by time. just use the extension point directly eg:

```
<completion.contributor language="PHP" implementationClass="fr.adrienbrault.idea.symfony2plugin.completion.ConstantEnumCompletionContributor"/>
```

Comment (skurfuerst) [inline src/main/java/de/vette/idea/neos/lang/yaml/references/nodeType/NodeTypeReference.java]: ah, got it, thanks a lot for explaining :) :-)

**Comment by skurfuerst** [comment]
fixed all comments :) Thanks for your reviews :heart:

**Pull request merged**